### PR TITLE
Update Web Browsers Section with drawbacks/benefits of Chrome, Firefox and Safari

### PR DESCRIPTION
**Pull request by Or3stis**
This is PR has significant changes over the original text.

Instead of proposing Chrome as the best option in terms of security, now the security and privacy benefits and drawbacks about Chrome, Firefox and Safari are presented. This way people can make an informed decision about their browser of choice.

Each browser has a separate section with references from official sources. I toned down some of the language and tried to remain impartial. Moreover, I added a paragraph about the increased attack surface that web extensions introduce and how each browser uses them.

Generic Privacy suggestions where moved to their own subsection (Web Browsers and Privacy)

Since the changes are quite significant, take your time to review them. If something is not clarified enough or needs additional references, I would be more than to happy to add them. I took quite a time proof reading this, but if I missed anything, please do not hesitate to correct me.

I understand that web browsers are a sensitive topic. Since we are talking about their security features, personal feelings should not interfere. The fact that someone loves a specific browser does not make it more secure. I would be more than happy to discuss such topics on a separate channel, but for the current PR, suggestions should be focussed in security and privacy and backed by data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
 
-Also be aware of [WebRTC](https://en.wikipedia.org/wiki/WebRTC#Concerns), which may reveal your local or public (if connected to VPN) IP address(es). This can be disabled with extensions such as [uBlock Origin](https://github.com/gorhill/uBlock/wiki/Prevent-WebRTC-from-leaking-local-IP-address) and [rentamob/WebRTC-Leak-Prevent](https://github.com/rentamob/WebRTC-Leak-Prevent).
+It is best to remember that Google is an advertising company and its major source of revenue is [AdSense](https://www.google.com/adsense/start/#/?modal_active=none). It makes perfect sense that an advertising company would leverage its services to maximize its profit. That means that while using [Google services](https://www.google.com/services/#?modal_active=none) certain personal information are being stored. Google is quite open about the data it stores and how it used them. Users can opt out from many of those services and see what type of information Google has stored from their [account settings](https://myaccount.google.com/privacy?pli=1).
```
Comment (ansell) [inline README.md]: "perfect sense" and "quite open" may not be the best way to put this. It makes the document look like an advertisement for using Google, where it is intended to be a neutral statement of facts (where even the best security facts are opinions in the end, but the neutrality should be front and center).

Replace the references to the Google sales pages, https://www.google.com/adsense/start/#/?modal_active=none and https://www.google.com/services/#?modal_active=none (has the page title "Google Business Solutions") and replace them with something more informative and not focused on Google's customers, rather being useful for Chrome users.

Changing to just "open" with a reference to where they state in a single document the list of data they store on Chrome users (in particular) would be useful.

File: README.md [hunk 2]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
 
-Many Chromium-derived browsers are not recommended. They are usually [closed source](http://yro.slashdot.org/comments.pl?sid=4176879&cid=44774943), [poorly maintained](https://plus.google.com/+JustinSchuh/posts/69qw9wZVH8z), [have bugs](https://code.google.com/p/google-security-research/issues/detail?id=679), and make dubious claims to protect privacy. See [The Private Life of Chromium Browsers](http://thesimplecomputer.info/the-private-life-of-chromium-browsers).
+#### Firefox
+
+[Firefox](https://www.mozilla.org/en-US/firefox/new/) is an excellent browser as well as being completely open source. Currently, Firefox is in a renaissance period. It replaces major parts of its infrastructure and code base under projects [Quantum](https://wiki.mozilla.org/Quantum) and [Photon](https://wiki.mozilla.org/Firefox/Photon/Updates). Part of the Quantum project is to replace C++ code with [Rust](https://www.rust-lang.org/en-US/). Rust is a systems programming with a focus on security and thread safety. It is expected that Rust adoption will greatly improve the overall security posture of Firefox.
```
Comment (ansell) [inline README.md]: "systems programming" to "systems programming language" or similar.

File: README.md [hunk 3]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+Firefox offers a similar security model to Chrome. It offers
```
Comment (ansell) [inline README.md]: "same as Chrome" to "similar to Chrome"

File: README.md [hunk 4]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+Previous versions of Firefox used a [Web Extension SDK](https://developer.mozilla.org/en-US/Add-ons/Legacy_add_ons) that was quite invasive and offered immense freedom to developers. Sadly, that freedom also introduced a number of vulnerabilities in Firefox that greatly affected its users. Currently, Firefox only supports Web Extensions through the [Web Extension Api](https://developer.mozilla.org/en-US/Add-ons/WebExtensions), which is very similar to Chrome's.
```
Comment (ansell) [inline README.md]: A reference for "greatly affected its users" would be great.

File: README.md [hunk 5]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+Firefox offers a similar security model to Chrome. It offers
+[bounty](https://www.mozilla.org/en-US/security/bug-bounty/) program, although it is not a lucrative as Chrome's. Firefox follows a six-week release cycle same as Chrome.
+
+See discussion in issues [#2](https://github.com/drduh/OS-X-Security-and-Privacy-Guide/issues/2), [#90](https://github.com/drduh/OS-X-Security-and-Privacy-Guide/issues/90) for more information about certain differences in Firefox and Chrome.
+
+If using Firefox, see [TheCreeper/PrivacyFox](https://github.com/TheCreeper/PrivacyFox) for recommended privacy preferences. Also be sure to check out [NoScript](https://noscript.net/) for Mozilla-based browsers, which allows whitelist-based, pre-emptive script blocking.
```
Comment (ansell) [inline README.md]: PrivacyBadger/HTTPSEverywhere/Cert Patrol/UBlock Origin are referenced in the Chrome section, but are also applicable here. Not sure if it is worth the duplication but just putting it out there.

File: README.md [hunk 6]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+Safari syncs user's preferences and stored logins through iCloud. Stored passwords are encrypted. In order to be viewed in plain text, a user must input the password of the current device. This means that users can sync data across devices with added security.
+
+Safari follows a slower release cycle than Chrome and Firefox (3-4 minor releases, 1 major release, per year). Newer features are slower to be adopted to the stable channel. Although security updates in Safari are handled independent of the stable release schedule and issued automatically through the App store. The Safari channel that follows a six-week release cycle (same as Chrome and Firefox) is called [Safari Technology Preview](https://developer.apple.com/safari/technology-preview/) and it is the recommended option instead of the stable channel of Safari.
```
Comment (ansell) [inline README.md]: "same as Chrome and Firefox" to "similar to Chrome and Firefox"

File: README.md [hunk 7]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
 
-Use [Google Chrome](https://www.google.com/chrome/browser/desktop/) for most of your browsing. It offers [separate profiles](https://www.chromium.org/user-experience/multi-profiles), [good sandboxing](https://www.chromium.org/developers/design-documents/sandbox), [frequent updates](http://googlechromereleases.blogspot.com/) (including Flash, although you should disable it - see below), and carries [impressive credentials](https://www.chromium.org/Home/chromium-security/brag-sheet).
+The best tip to ensure secure browsing regardless your choice of Web Browser is proper security hygiene. The majority of Web Browser exploits require social engineering attacks to achieve native code execution. Always be mindful of the links you click and be extra careful when websites ask you to download and install external software. 99% percent of the time that software is malware.
```
Comment (ansell) [inline README.md]: Remove "external", as it isn't clear what it means in this case.

File: README.md [hunk 8]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+Previous versions of Firefox used a [Web Extension SDK](https://developer.mozilla.org/en-US/Add-ons/Legacy_add_ons) that was quite invasive and offered immense freedom to developers. Sadly, that freedom also introduced a number of vulnerabilities in Firefox that greatly affected its users. Currently, Firefox only supports Web Extensions through the [Web Extension Api](https://developer.mozilla.org/en-US/Add-ons/WebExtensions), which is very similar to Chrome's.
+
+Submission of Web Extensions is Firefox is free. Web Extensions is Firefox most of the time are Open Source, although certain Web Extensions are proprietary.
```
Comment (ansell) [inline README.md]: "Web Extensions is" to "Web Extensions in"

File: README.md [hunk 9]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+#### Safari
+
+[Safari](https://www.apple.com/safari/) is the default Web Browser of macOS. It is also the best browser regarding battery performance. Safari, like Chrome, has both Open Source and proprietary components. Safari is based on the open source Web Engine [WebKit](https://en.wikipedia.org/wiki/WebKit), which is ubiquitous among the macOS ecosystem. WebKit is used by Apple apps such as Mail, iTunes, iBooks, and the App store. Chrome's [Blink](https://www.chromium.org/blink) engine is a fork of WebKit and both engines share a number of similarities.
+
+Safari supports certain unique features that benefit user security and privacy. [Content blockers](https://webkit.org/blog/3476/content-blockers-first-look/) enables the creation of content blocking rules without using Javascript. This rule based approach greatly improves memory user, security, and privacy. Safari 11 will introduce an [Intelligent Tracking Prevention](https://webkit.org/blog/7675/intelligent-tracking-prevention/) system. This feature will automatically remove tracking data stored in Safari after a period noninteraction by the user from the tracker's website.
```
Comment (ansell) [inline README.md]: "period noninteraction" to "period of non-interaction"

File: README.md (hunk 7 above)
Comment (Or3stis) [reply README.md]: done

File: README.md [hunk 10]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
 
-Chrome also comes with a great [PDF viewer](http://0xdabbad00.com/2013/01/13/most-secure-pdf-viewer-chrome-pdf-viewer/).
+Another important consideration about Web Browser security is Web Extensions. Web Extensions greatly increase the attack surface of the Web Browser. This is an issue that plagues Firefox and [Chrome](https://encrypted.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=3&cad=rja&uact=8&ved=0ahUKEwitnsiktvTVAhWpAsAKHUa1B3EQFggyMAI&url=https:%2F%2Fcourses.csail.mit.edu%2F6.857%2F2016%2Ffiles%2F24.pdf&usg=AFQjCNHZLw9aoHFwPTt020U56MLjRhNMlQ) alike. Luckily, Web Extensions can only access specific browser APIs that are being governed by their manifest. That means we can quickly audit their behavior and remove them if they request access to information they shouldn't (why would an Ad blocker require camera access?). In the interest of security, it is best to limit your use of Web Extensions.
```
[the remark this answers is not in the document]
Comment (Or3stis) [reply README.md]: done

**Comment by Or3stis** [comment]
Thanks @ansell for your helpful comments.

I have addressed all of them in the present commit 😄 

Please, let me know, if you have other suggestions.

File: README.md [hunk 11]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+Chrome offers account sync between multiple devices. Part of the sync data are stored website logins. The login passwords are encrypted and in order to access them, a user's Google account password is required. You can use your Google account to sign to your Chrome customized settings from other devices while retaining your the security of your passwords.
 
-Create at least three profiles, one for browsing **trusted** Web sites (email, banking), another for **mostly trusted** Web sites (link aggregators, news sites), and a third for a completely **cookie-less** and **script-less** experience.
+Chrome's Web store for extensions requires a [$5 dollar lifetime fee](https://developer.chrome.com/webstore/publish#pay-the-developer-signup-fee)in order to submit extensions. The low cost allows the development of many quality Open Source Web Extensions that do not aim to monetize through usage.
```
Comment (drduh) [inline README.md]: Space after URL here.

File: README.md [hunk 12]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+  1. Adobe Flash Plugin. Google Chrome supports a Pepper API version of Adobe Flash which gets updated automatically with Chrome.
+  1. Media Codec support. Adds support for proprietary codecs.
+  1. Chrome's [PDF viewer](http://0xdabbad00.com/2013/01/13/most-secure-pdf-viewer-chrome-pdf-viewer/).
+
+Chrome offers [separate profiles](https://www.chromium.org/user-experience/multi-profiles), [sandboxing](https://www.chromium.org/developers/design-documents/sandbox), [frequent updates](http://googlechromereleases.blogspot.com/) (including Flash, although you should disable it - see below), and carries [impressive credentials](https://www.chromium.org/Home/chromium-security/brag-sheet). In addition, Google offers a very lucrative [bounty](https://www.google.com/about/appsecurity/chrome-rewards/) program for reporting vulnerabilities along with its own [Project Zero](googleprojectzero.blogspot.com). This means that a large number of highly talented and motivated people are constantly auditing Chrome's code base.
```
Comment (drduh) [inline README.md]: Fully qualify projectzero URL.

File: README.md [hunk 13]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
 
-Create at least three profiles, one for browsing **trusted** Web sites (email, banking), another for **mostly trusted** Web sites (link aggregators, news sites), and a third for a completely **cookie-less** and **script-less** experience.
+Chrome's Web store for extensions requires a [$5 dollar lifetime fee](https://developer.chrome.com/webstore/publish#pay-the-developer-signup-fee)in order to submit extensions. The low cost allows the development of many quality Open Source Web Extensions that do not aim to monetize through usage.
+
+Chrome has the largest share of global usage and is the preferred target platform for the majority of developers. Major technologies are based on Chrome's Open Source components, such as [node.js](https://nodejs.org/en/) which uses [Chrome's V8](https://developers.google.com/v8/) Engine and the [Electron](https://electron.atom.io/) framework, which is based on Chromium and node.js. Chrome's vast user base makes it the most attractive target for threat actors. Despite under constants attacks, Chrome has retained an impressive security track record over the years. This is not a small feat.
```
Comment (drduh) [inline README.md]: ... attractive target for threat actors *and security researchers* ... perhaps?

File: README.md [hunk 14]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+To improve your privacy and security posture, create at least three profiles, one for browsing **trusted** Web sites (email, banking), another for **mostly trusted** Web sites (link aggregators, news sites), and a third for a completely **cookie-less** and **script-less** experience.
```
Comment (drduh) [inline README.md]: ... security posture *of the browser* ...

File: README.md [hunk 15]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+**An important security consideration about Firefox**. Similar to Chrome and Safari, Firefox allows account sync across multiple devices. While stored login passwords are encrypted, Firefox does not require a password to reveal their plain text format. Firefox only displays as yes/no prompt. This is an important security issue. Keep that in mind if you sign in to your Firefox account from devices that do not belong to you and leave them unattended. The [issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1393493) has been raised among the Firefox community and hopefully will be resolved in the coming versions.
```
Comment (drduh) [inline README.md]: If you're going to use bold, I'd prefer one word, like **Note** or **Warning**

File: README.md [hunk 16]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+#### Safari
+
+[Safari](https://www.apple.com/safari/) is the default Web Browser of macOS. It is also the best browser regarding battery performance. Safari, like Chrome, has both Open Source and proprietary components. Safari is based on the open source Web Engine [WebKit](https://en.wikipedia.org/wiki/WebKit), which is ubiquitous among the macOS ecosystem. WebKit is used by Apple apps such as Mail, iTunes, iBooks, and the App store. Chrome's [Blink](https://www.chromium.org/blink) engine is a fork of WebKit and both engines share a number of similarities.
```
Comment (drduh) [inline README.md]: ... It is also the most optimized browser for reducing battery use ..., perhaps?

File: README.md [hunk 17]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+Safari supports certain unique features that benefit user security and privacy. [Content blockers](https://webkit.org/blog/3476/content-blockers-first-look/) enables the creation of content blocking rules without using Javascript. This rule based approach greatly improves memory user, security, and privacy. Safari 11 will introduce an [Intelligent Tracking Prevention](https://webkit.org/blog/7675/intelligent-tracking-prevention/) system. This feature will automatically remove tracking data stored in Safari after a period of non-interaction by the user from the tracker's website.
+
+Similar to Chrome and Firefox, Safari offers a [bounty program](https://developer.apple.com/bug-reporting/) for bug reporting, that offers substantial rewards to security researchers.
```
Comment (drduh) [inline README.md]: This bug reporting page makes no mention of "substantial rewards". Either remove the qualifier or link to their VRP page (if there's one).

File: README.md [hunk 18]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+Safari syncs user's preferences and stored logins through iCloud. Stored passwords are encrypted. In order to be viewed in plain text, a user must input the password of the current device. This means that users can sync data across devices with added security.
```
Comment (drduh) [inline README.md]: Any more information on how stored passwords are encrypted?

File: README.md [hunk 19]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+All Web Browsers retain certain information about our browsing habits. That information is used for a number of reasons. One of them is to improve the overall performance of the Web Browser. Most Web Browsers offer predictions services to resolve typos or URL redirections, store analytics data of browsing patterns, crash reports and black listing of known malicious servers. Those options can be turned on and off from each Web Browser's settings panel.
+
+Since Web Browser executes untrusted code from the server, it is important to understand what type of information can be accessed. The [Navigator](https://developer.mozilla.org/en-US/docs/Web/API/Navigator) interface gives access to information about the Web Browsers user agent. Those include information such as the operating system, Websites' permissions, and the device's battery level. For more information about security conscious browsing and what type of information is being "leaked" by your browser, see [HowTo: Privacy & Security Conscious Browsing](https://gist.github.com/atcuno/3425484ac5cce5298932), [browserleaks.com](https://www.browserleaks.com/) and [EFF Panopticlick](https://panopticlick.eff.org/).
```
Comment (drduh) [inline README.md]: Web **browsers**

File: README.md [hunk 20]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+Disable third-party cookies from your Web Browser settings.
```
Comment (drduh) [inline README.md]: Why; what effect does this have and why is it practical (or not)?

File: README.md [hunk 21]
```diff
@@ -1030,41 +1034,105 @@ You can replace ad images with pictures of kittens, for example, by starting the
...
+
+A great step to improve privacy is to replace your default search engine to [duckduckgo](https://duckduckgo.com/), which is committed to user privacy. Moreover, it offers some unique features, such as color Themes and direct site search through the use of [bangs](https://duckduckgo.com/bang).
```
Comment (drduh) [inline README.md]: I think this is outside the scope of this guide.

Comment (Or3stis) [reply README.md]: I included duckduckgo since the guide is about privacy as well as security.

It can be removed.

File: README.md (hunk 20 above)
Comment (Or3stis) [reply README.md]: The link about the navigator is to offer readers additional information. Most people do not know that a web browser has access to the battery level of a device and so on.

I could add a bit more clarity to the paragraph.

File: README.md (hunk 18 above)
Comment (Or3stis) [reply README.md]: I don't know the specifics, but I can look it up.

File: README.md (hunk 17 above)
Comment (Or3stis) [reply README.md]: The Apple's bug bounty program is invite only. That means that unless you have an invited developer account you won't get rewarded for reporting bugs. They announced the bounty program a couple of years ago during the BlackHat convention. Official mention of cash prizes of cash prizes is shown in the end of their [presentation](https://www.blackhat.com/docs/us-16/materials/us-16-Krstic.pdf).

I update the document.

File: README.md (hunk 16 above)
Comment (Or3stis) [reply README.md]: Do you want the sentence rephrased to be more neutral or removed?

File: README.md (hunk 15 above)
Comment (Or3stis) [reply README.md]: fixed

File: README.md (hunk 14 above)
Comment (Or3stis) [reply README.md]: fixed

File: README.md (hunk 13 above)
Comment (Or3stis) [reply README.md]: fixed

File: README.md (hunk 11 above)
Comment (Or3stis) [reply README.md]: fix

File: README.md (hunk 12 above)
Comment (Or3stis) [reply README.md]: What do you mean?

File: README.md (hunk 19 above)
Comment (Or3stis) [reply README.md]: fixed

File: README.md (hunk 17 above)
Comment (Or3stis) [reply README.md]: fixed

**Comment by Or3stis** [comment]
@drduh I addressed most of your comments.

I left the first one, which confused me and the last two that can be removed or clarified.

File: README.md (hunk 12 above)
Comment (drduh) [reply README.md]: Use https://googleprojectzero.blogspot.com; without a URI, it thinks the path is relative.

File: README.md (hunk 20 above)
Comment (drduh) [reply README.md]: Sorry, the comment was asking why disabling third-party cookies is a good idea.

File: README.md (hunk 21 above)
Comment (drduh) [reply README.md]: True, but I would rather not get into recommendations which do not directly pertain to the OS; please remove.

File: README.md (hunk 18 above)
Comment (drduh) [reply README.md]: Well, at least verify they're indeed encrypted.

**Comment by drduh** [comment]
Just a few more comments to address above and this is ready to merge.

File: README.md (hunk 18 above)
Comment (Or3stis) [reply README.md]: fixed

File: README.md (hunk 20 above)
Comment (Or3stis) [reply README.md]: fixed

I added a quick explanation about third party cookies

File: README.md (hunk 21 above)
Comment (Or3stis) [reply README.md]: removed

**Comment by Or3stis** [comment]
@drduh I addressed the rest of the comments 😄 

Have a final look and we are ready for merge 🎉

**Comment by drduh** [comment]
Thank you, @Or3stis for raising the issue and submitting an excellent fix.

**Pull request merged**